packages/fx-core/src/core/callback.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { CoreCallbackEvent, CoreCallbackFunc } from "@microsoft/teamsfx-api";
+
+export class CallbackRegistry {
+  private static registry: Map<CoreCallbackEvent, CoreCallbackFunc[]> = new Map();
+
+  public static has(event: CoreCallbackEvent): boolean {
+    return this.registry.has(event);
+  }
+
+  public static set(event: CoreCallbackEvent, func: CoreCallbackFunc): void {
+    if (!this.registry.has(event)) {
+      this.registry.set(event, []);
+    }
+    const funcs = this.registry.get(event) as CoreCallbackFunc[];
+    funcs.push(func);
+    this.registry.set(event, funcs);
```

Comment:
In corner case, if the event is registered twice, the map will include duplicated callbacks and the callback will be called twice when the event fired.

Reply:
By design. I already test this kind of case in UT